app/code/Magento/InventoryCatalog/Model/ResourceModel/AddStockDataToCollection.php

```diff
@@ -68,16 +79,27 @@ public function execute(Collection $collection, bool $isFilterInStock, int $stoc
             );
             $isSalableColumnName = IndexStructure::IS_SALABLE;
             $collection->getSelect()
-                ->join(
+                ->joinLeft(
                     ['stock_status_index' => $stockIndexTableName],
                     'product.sku = stock_status_index.' . IndexStructure::SKU,
                     [$isSalableColumnName]
                 );
         }
 
         if ($isFilterInStock) {
+            $globalManageStock = (int)$this->stockConfiguration->getManageStock();
+            $stockSettingsCondition = '(legacy_stock_item.use_config_manage_stock = 0 AND legacy_stock_item.manage_stock = 0)';
+            if (0 === $globalManageStock) {
+                $stockSettingsCondition .= ' OR legacy_stock_item.use_config_manage_stock = 1';
+            }
```

Comment:
You can use here
\Magento\InventorySales\Model\ResourceModel\IsStockItemSalableCondition\ManageStockCondition